lib/client-credentials.js

```diff
@@ -0,0 +1,87 @@
+'use strict'
+
+const Client = require('../lib/client')
+const Token = require('../lib/token')
+const Promise = require('bluebird')
+const uuid = require('uuid')
+
+module.exports = {
+  getAccessToken: function (bearerToken) {
+    var token = null
+
+    return Token.objects.get({
+      access_token: bearerToken
+    })
+    .then(function (_token) {
+      token = _token
+      return token.client
+    })
+    .then(function (client) {
+      return {
+        accessToken: token.access_token,
+        client: client,
+        accessTokenExpiresAt: token.access_expires,
+        refreshTokenExpiresAt: token.refresh_expires,
+        user: {
+          email: token.user_email
+        }
+      }
+    })
+  },
+
+  getClient: function (clientId, clientSecret) {
+    return Client.objects.get({
+      client_id: clientId,
+      client_secret: clientSecret
+    })
+    .then(function (client) {
+      return {
+        clientId: client.client_id,
+        clientSecret: client.client_secret,
+        grants: ['client_credentials']
+      }
+    })
+  },
+
+  getUserFromClient: function (client) {
+    return Client.objects.get({
+      client_id: client.clientId,
+      client_secret: client.clientSecret
+    }).then(function (client) {
+      return {
+        email: client.email
+      }
+    })
+  },
+
+  saveToken: function (token, client, user) {
```

Comment:
(for the future: `saveToken (token, client, user) { }` is a shorthand for `saveToken: function (token, client, user) { }` in Node v4+.)